### PR TITLE
[SofaCommon] Pluginize SofaExplicitOdeSolver

### PR DESCRIPTION
**Pull request by fredroy**
#1597

- Usual cleaning (pragma, namespaces)
- warnings removal
- cmake dep polish
- remove useless includes

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp [hunk 1]
```diff
@@ -458,7 +458,7 @@ std::map<std::string, ComponentChange> uncreatableComponents = {
 
     // SofaEngine was pluginized in #1603
     { "BoxROI", Pluginized("v20.12", "SofaEngine") },
-
+
```
Comment (epernod) [inline SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp:461]: info regarding EulerSolver pluginized  is missing

File: SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/RayTriangleIntersection.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@
 ******************************************************************************/
 #include <SofaMeshCollision/RayTriangleIntersection.h>
 
-#include <SofaMeshCollision/TriangleModel.h>
+#include <SofaMeshCollision/Triangle.h>
```
Comment (epernod) [inline SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/RayTriangleIntersection.cpp]: again same as previous PR :)

File: SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/initSofaMeshCollision.cpp [hunk 3]
```diff
@@ -64,7 +64,7 @@ const char* getModuleName()
 
 const char* getModuleVersion()
 {
-    return sofa_tostring(SOFACONSTRAINT_VERSION);
+    return sofa_tostring(SOFAMESHCOLLISION_VERSION);
```
Comment (epernod) [inline SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/initSofaMeshCollision.cpp:67]: strange... why in this PR...

Comment (fredroy) [reply SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/initSofaMeshCollision.cpp:67]: chuttt 😗

File: SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/RayTriangleIntersection.cpp (hunk 2 above)
Comment (fredroy) [reply SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/RayTriangleIntersection.cpp]: argl

File: SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp (hunk 1 above)
Comment (fredroy) [reply SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp:461]: 👍

**Pull request merged**